configs/rails_config.yml

```diff
@@ -110,27 +116,35 @@ jobs:
       - run:
           name: "Compile and serve assets"
           command: |
-            if [ $CIRCLE_BRANCH == staging ]
-            then
-              export ASSETS_ENV=staging
-              export S3_ASSETS_BUCKET=<< pipeline.parameters.assets-bucket-staging >>
-            elif [ $CIRCLE_BRANCH == master ]
-            then
-              export ASSETS_ENV=production
-              export S3_ASSETS_BUCKET=<< pipeline.parameters.assets-bucket-production >>
-            else
-              exit 0
+            branches=<< pipeline.parameters.branches-for-assets >>
+            assetsBuckets=<< pipeline.parameters.assets-buckets >>
+            if [ ${#assetsBuckets[@]} -eq 0 ]; then
+              assetsBuckets=('<< pipeline.parameters.assets-bucket-production >>' '<< pipeline.parameters.assets-bucket-staging >>')
             fi
 
-            yarn
-            RAILS_ENV=${ASSETS_ENV} bundle exec rake db:create db:migrate
-            RAILS_ENV=${ASSETS_ENV} bundle exec rake assets:clobber assets:precompile
-            aws s3 cp public/assets/ s3://${S3_ASSETS_BUCKET}/assets/ --recursive --acl public-read
+            for i in "${!branches[@]}"; do
+              if [ $CIRCLE_BRANCH == ${branches[$i]%%:*} ]; then
+                export ASSETS_ENV=${branches[$i]##*:}
+                export S3_ASSETS_BUCKET=${assetsBuckets[$i]}
```

Comment:
seul le dernier bucket sera pris en compte dans la var d'env ici non ? ça marche pour ASSETS_ENV j'imagine comme il y à l'air d'avoir `:` à la fin. Ça fait quoi %%:* et ##*: ?

Reply:
Ici on entre dans la condition ou la branche sur laquele circleCi fait son check correspond a une des branches indiquées dans ta liste  `branches-for-assets`. Ca ne peut donc arriver qu'une seule fois.
`$i` correspond a l'index de cette branche dans ta liste (c'est le `i` de `for i in "${!branches[@]}"`). Donc pour trouver le bucket correspondant je regarde juste dans la liste des buckets a la meme position.

Reply:
pour répondre a l'autre question sur `%%` et `##` ca me sert a extraire des éléments de la string jusqu'a / depuis un élément (ici c'est le `:`), comme ca dans la liste des branches je peux indiquer `[Branch_name]:[Environment]` ou juste `Branch_name` et ca fonctionne de la meme maniere. C'est pratique pour la branche master qui a un nom d'env différent.

L'explication est la (https://stackoverflow.com/questions/918886/how-do-i-split-a-string-on-a-delimiter-in-bash):

```markdown
${var#*SubStr}  # drops substring from start of string up to first occurrence of `SubStr`
${var##*SubStr} # drops substring from start of string up to last occurrence of `SubStr`
${var%SubStr*}  # drops substring from last occurrence of `SubStr` to end of string
${var%%SubStr*} # drops substring from first occurrence of `SubStr` to end of string
As explained by Score_Under:

# and % delete the shortest possible matching substring from the start and end of the string respectively, and

## and %% delete the longest possible matching substring.

Using the above syntax, we can create an approach where we extract substring "elements" from the string by deleting the substrings up to or after the delimiter.

The codeblock below works well in bash (including Mac OS's bash), dash, ksh, and busybox's ash:

(Thanks to Adam Katz's comment, making this loop a lot simplier!)

IN="bla@some.com;john@home.com;Full Name <fulnam@other.org>"
while [ "$IN" != "$iter" ] ;do
    # extract the substring from start of string up to delimiter.
    iter=${IN%%;*}
    # delete this first "element" AND next separator, from $IN.
    IN="${IN#$iter;}"
    # Print (or doing anything with) the first "element".
    echo "> [$iter]"
done
# > [bla@some.com]
# > [john@home.com]
# > [Full Name <fulnam@other.org>]
```